src/main/java/model/Cars.java

```diff
@@ -0,0 +1,73 @@
+package model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static util.Util.generateRandomNumber;
+
+public class Cars {
+
+    private static final int MIN_CAR_NAME_COUNT = 2;
+    private static final int MIN_POSITION = 0;
+
+    private final List<Car> cars;
+
+    private Cars(final List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public static Cars from(final List<String> names) {
+        validateNames(names);
+
+        List<Car> cars = names.stream()
+                .map(Car::from)
+                .collect(Collectors.toList());
+
+        return new Cars(cars);
+    }
+
+    private static void validateNames(List<String> names) {
+        checkDuplicatedName(names);
+        checkNameCount(names);
+    }
+
+    private static void checkDuplicatedName(List<String> names) {
+        long nameCount = names.stream().distinct().count();
+        if (names.size() != nameCount) {
+            throw new IllegalArgumentException("[ERROR] 자동차의 이름은 중복될 수 없습니다.");
+        }
+    }
+
+    private static void checkNameCount(List<String> names) {
+        if (names.size() < MIN_CAR_NAME_COUNT) {
+            throw new IllegalArgumentException("[ERROR] 자동차 이름은 2개 이상이어야 합니다.");
+        }
+    }
+
+    public void moveCars() {
+        for (Car car : this.cars) {
+            car.moveForward(generateRandomNumber());
+        }
+    }
+
+    public List<String> findMaxPositionCarName() {
+        int maxPosition = calculateMaxPosition();
+
+        return this.cars.stream()
+                .filter(car -> car.getPosition() == maxPosition)
+                .map(Car::getName)
+                .collect(Collectors.toList());
```

Comment:
```suggestion
        return this.cars.stream()
                .filter(car -> car.getPosition() == maxPosition)
                .map(Car::getName)
                .toList();
```
Java 16 부터는 이런 메서드도 사용할 수 있어요
그런데 `collect(Collectors.toList())` 와 `toList()` 에는 차이점이 있는데요,
한 번 확인해보셔도 좋을 것 같아요!

Reply:
https://binux.tistory.com/146
검색해 보니 위 블로그가 떠서 확인해 보았습니다!
`collect(Collectors.toList())`는 ArrayList를 반환하고,
`toList()`는 UnmodifiableList를 반환합니다.

Cars는 불변 객체이기 때문에 toList로 변경하면 더 좋을 것 같습니다!

Reply:
수정해주시긴 했는데 막상 리뷰 남긴 `findMaxPositionCarName()` 쪽 collect 는 변경이 안됐네요 :)